src/Tasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -567,30 +567,20 @@ internal static string GetRuntimeVersion(string path)
 #if FEATURE_MSCOREE
             if (NativeMethodsShared.IsWindows)
             {
-                StringBuilder runtimeVersion;
-                uint hresult;
-#if DEBUG
-                // Just to make sure and exercise the code that doubles the size
-                // every time GetRequestedRuntimeInfo fails due to insufficient buffer size.
-                int bufferLength = 1;
-#else
-                int bufferLength = 11; // 11 is the length of a runtime version and null terminator v2.0.50727/0
-#endif
-                do
-                {
-                    runtimeVersion = new StringBuilder(bufferLength);
-                    hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out _);
-                    bufferLength *= 2;
-                } while (hresult == NativeMethodsShared.ERROR_INSUFFICIENT_BUFFER);
-
-                if (hresult == NativeMethodsShared.S_OK)
+                unsafe
                 {
-                    return runtimeVersion.ToString();
-                }
-                else
-                {
-                    return String.Empty;
-                }
+                    // Run the first GetFileVersion to get the required buffer size.
+                    int bufferLength = 1;
+                    uint hresult = NativeMethods.GetFileVersion(path, null, bufferLength, out int dwLength);
```

Comment:
nit: Please check the `hresult` here. If it's not `ERROR_INSUFFICIENT_BUFFER` `dwLength` may be bogus/undefined and we should not continue.